File: src/game/entity/blocks/InvisibleBlock.java
package game.entity.blocks;

import game.entity.models.Entity;
import game.entity.models.Coordinates;

import java.awt.image.BufferedImage;

public class InvisibleBlock extends HardBlock{

    public InvisibleBlock(Coordinates coordinates){
        super(coordinates);
    }

    @Override
    protected void doInteract(Entity e) {}

    @Override
    public BufferedImage getImage() {
        return null;
    }
}

File: src/game/entity/blocks/InvisibleBlock.kt
package game.entity.blocks

import game.entity.EntityTypes
import game.entity.models.Coordinates
import game.entity.models.Entity
import java.awt.image.BufferedImage

class InvisibleBlock : HardBlock {
    constructor(id: Long) : super(id)
    constructor(coordinates: Coordinates?) : super(coordinates)

    override fun doInteract(e: Entity?) {}

    override fun getImage(): BufferedImage? = null

    override val type: EntityTypes = EntityTypes.InvisibleBlock
}
